fe/fe-core/src/main/java/org/apache/doris/nereids/analyzer/UnboundFunction.java

```diff
@@ -77,4 +77,24 @@ public <R, C> R accept(ExpressionVisitor<R, C> visitor, C context) {
     public Expression withChildren(List<Expression> children) {
         return new UnboundFunction(name, isDistinct, children);
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        if (!super.equals(o)) {
+            return false;
+        }
+        UnboundFunction that = (UnboundFunction) o;
+        return isDistinct == that.isDistinct && name.equals(that.name);
```

Comment:
I'm not sure whether `equals` of `Expression` should use children `equals()`.

`Equal (1, 2)` not equals to ``Equal(1, 3)`.